shuffle-coordinator/src/main/java/com/alibaba/flink/shuffle/coordinator/metrics/MetricsRestHandler.java

```diff
@@ -0,0 +1,160 @@
+import com.alibaba.metrics.FastCompass;
+import com.alibaba.metrics.Gauge;
+import com.alibaba.metrics.Histogram;
+import com.alibaba.metrics.IMetricManager;
+import com.alibaba.metrics.Meter;
+import com.alibaba.metrics.MetricManager;
+import com.alibaba.metrics.MetricName;
+import com.alibaba.metrics.MetricRegistry;
+import com.alibaba.metrics.Timer;
+import com.alibaba.metrics.common.CollectLevel;
+import com.alibaba.metrics.common.MetricObject;
+import com.alibaba.metrics.common.MetricsCollector;
+import com.alibaba.metrics.common.MetricsCollectorFactory;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A rest handler for metrics request.
+ *
+ * <p>This class borrows code from com.alibaba.metrics.rest.MetricsResource.
+ */
+public class MetricsRestHandler implements RestHandler {
+
+    public static final String QUERY_PATH = "/metrics";
+
+    public static final String PARAM_GROUP = "group";
+
+    public static final String ALL_REMOTE_SHUFFLE = "remote-shuffle";
+
+    public static final String REMOTE_SHUFFLE_PREFIX = "remote-shuffle.";
+
+    @Override
+    public RestResponse handle(Map<String, String[]> parameters) {
+        // currently, we support all metrics and group metrics query
+        IMetricManager metricManager = MetricManager.getIMetricManager();
+        CommonUtils.checkState(metricManager.isEnabled(), "Metrics are not enabled.");
+
+        List<String> allGroups = metricManager.listMetricGroups();
+        final Set<String> distinctGroups = new HashSet<>();
+        String[] queryGroups = parameters.get(PARAM_GROUP);
+
+        if (queryGroups != null) {
+            distinctGroups.addAll(Arrays.asList(queryGroups));
+        }
+
+        if (distinctGroups.contains(ALL_REMOTE_SHUFFLE)) {
+            distinctGroups.remove(ALL_REMOTE_SHUFFLE);
+            allGroups.stream()
+                    .filter(group -> group.startsWith(REMOTE_SHUFFLE_PREFIX))
+                    .forEach(distinctGroups::add);
+        }
+
+        Map<String, List<MetricObject>> metricsData = new TreeMap<>();
+        if (distinctGroups.isEmpty()) {
+            allGroups.stream()
+                    .filter(group -> !group.equals("tomcat") && !group.equals("nginx"))
+                    .forEach(distinctGroups::add);
+        }
+
+        for (String queryGroup : distinctGroups) {
+            if (allGroups.contains(queryGroup)) {
+                MetricRegistry metricRegistry = metricManager.getMetricRegistryByGroup(queryGroup);
+                metricsData.put(queryGroup, buildMetrics(metricRegistry));
+            }
+        }
+
+        return new MetricResponse(metricsData);
+    }
+
+    @Override
+    public String getQueryPath() {
+        return QUERY_PATH;
+    }
+
+    private List<MetricObject> buildMetrics(MetricRegistry registry) {
```

Comment:
Why not use `MetricManager.getIMetricManager().getAllCategoryMetrics(MetricFilter.ALL);`  to get all metrics in one line?

Reply:
The current implementation borrows code from alibaba metrics (com.alibaba.metrics.rest.MetricsResource) and it has some rich information which I think is better. Here is an example:
    {
      "interval": 15,
      "metric": "remote-shuffle.storage.num_available_reading_buffers",
      "metricLevel": "NORMAL",
      "metricType": "GAUGE",
      "tags": {
        
      },
      "timestamp": 1639044931505,
      "value": 131072
    }